### PR TITLE
Fix inspect of Series containing NaN

### PR DESCRIPTION
**Pull request by philss**
Closes https://github.com/elixir-nx/explorer/issues/322

PS: this was inspired in the encoding of dates and datetime. @joshuataylor, if you could take a look would be great! Thanks!

### CODE REVIEW

**Review by josevalim — APPROVED** [review]
LGTM but @cigrainger should review the Rust code :)

File: native/explorer/src/datatypes.rs [hunk 1]
```diff
@@ -380,7 +380,37 @@ fn encode_datetime_series<'b>(s: &Series, time_unit: TimeUnit, env: Env<'b>) ->
...
+        .downcast_iter()
+        .flat_map(move |iter| {
+            iter.into_iter().map(move |opt_v| {
+                opt_v
+                    .copied()
+                    .map(|x| {
+                        if x.is_finite() {
+                            x.encode(env)
+                        } else {
+                            match (x.is_nan(), x.is_sign_negative()) {
```
Comment (joshuataylor) [inline native/explorer/src/datatypes.rs:402]: If you put the `nan`, `neg_infinity`, `infinity` encode part at the top of the function so it doesn't have to encode each time, as these don't change.

**Review by joshuataylor — COMMENTED** [review]
Looks great! One note about encoding.

File: native/explorer/src/datatypes.rs (hunk 1 above)
Comment (philss) [reply native/explorer/src/datatypes.rs:406]: Oh, nice! I changed the code to fix that. Thank you!

File: native/explorer/src/datatypes.rs [hunk 2]
```diff
@@ -4,7 +4,7 @@ use rustler::resource::ResourceArc;
 use rustler::{Atom, Encoder, Env, NifStruct, Term};
 use std::convert::TryInto;
 
-use crate::atoms;
+use crate::atoms::{self, infinity, nan, neg_infinity};
 use crate::atoms::{calendar, calendar_atom, day, hour, microsecond, minute, month, second, year};
```
Comment (cigrainger) [inline native/explorer/src/datatypes.rs:8]: Do we want to combine these two use declarations?

**Review by cigrainger — APPROVED** [review]
Looks great!

File: native/explorer/src/datatypes.rs (hunk 2 above)
Comment (philss) [reply native/explorer/src/datatypes.rs]: I changed that to one `use` declaration. Thanks!

**Pull request merged**